packages/connection-form/src/components/advanced-options-tabs/authentication-tab/authentication-oidc.tsx

```diff
@@ -0,0 +1,167 @@
+          <FormFieldContainer>
+            <TextInput
+              data-testid="connection-oidc-auth-code-flow-redirect-uri-input"
+              onChange={({
+                target: { value },
+              }: React.ChangeEvent<HTMLInputElement>) => {
+                handleFieldChanged('redirectURI', value);
+              }}
+              optional
+              label="Auth Code Flow Redirect URI"
+              value={connectionOptions?.oidc?.redirectURI || ''}
+            />
+          </FormFieldContainer>
+
+          <FormFieldContainer>
+            <Checkbox
+              onChange={({
+                target: { checked },
+              }: React.ChangeEvent<HTMLInputElement>) => {
+                if (checked) {
+                  updateConnectionFormField({
+                    type: 'update-auth-mechanism-property',
+                    key: 'ALLOWED_HOSTS',
+                    value: '*',
+                  });
+                  return;
+                }
+
+                // Unset the allowed hosts.
+                updateConnectionFormField({
+                  type: 'update-auth-mechanism-property',
+                  key: 'ALLOWED_HOSTS',
+                  value: '',
+                });
+              }}
+              data-testid="oidc-allow-untrusted-endpoint-input"
+              id="oidc-allow-untrusted-endpoint-input"
+              label={
+                <>
+                  <Label htmlFor="oidc-allow-untrusted-endpoint-input">
+                    Enable untrusted target endpoint
+                  </Label>
+                  <Description>
+                    Allow connecting when the target endpoint is not in the list
+                    of trusted endpoints &#40;this sets the driver&apos;s
+                    ALLOWED_HOSTS list to *&#41;
+                  </Description>
+                </>
+              }
+              checked={hasEnabledUntrustedEndpoints}
+            />
+          </FormFieldContainer>
+
+          <FormFieldContainer>
+            <Checkbox
+              onChange={({
+                target: { checked },
+              }: React.ChangeEvent<HTMLInputElement>) => {
+                if (checked) {
+                  handleFieldChanged('allowedFlows', ['device-auth']);
+                  return;
+                }
+
+                const newAllowedFlows = (
+                  connectionOptions.oidc?.allowedFlows as AuthFlowType[]
+                )?.filter?.((allowedFlow) => allowedFlow !== 'device-auth');
+                handleFieldChanged(
+                  'allowedFlows',
+                  newAllowedFlows.length > 0 ? newAllowedFlows : undefined
+                );
+              }}
+              data-testid="oidc-enable-device-auth-flow-input"
+              id="oidc-enable-device-auth-flow-input"
+              label={
+                <Label htmlFor="oidc-enable-device-auth-flow-input">
+                  Enable device authentication flow
```

Comment:
I’ve usually seen this capitalized because it’s a proper noun, not sure if that looks weird:

```suggestion
                  Enable Device Authentication Flow
```

For this one we’ll definitely want a description, e.g. “Less secure authentication flow that can be used as a fallback when browser-based authentication is unavailable”.

This is probably extremely rare for Compass users because we’re already in a GUI environment to begin with, so I’d maybe consider putting this behind a setting similar to what we do for the Kerberos password field (but that doesn’t have to be part of this PR).

Reply:
Nice catch, the standard in other tabs is capitalized. Sounds good, lets visit that in that ticket.